### DISCUSSION TITLE
Increase API request size limit

### DISCUSSION THREAD

**Discussion by josemalm32** [infinispan/infinispan#10452] [Q&A] [answered]
Hello, thank you for this great product. But I have a little problem and I can't find a solution. I have a request to cache about 11 MB in JSON. When I try to cache this request I get an error 413 "Request Entity Too Large". Is there any way to increase the limit? I have read the documentation, but I can't find anything. 

Can you give me a help? Thank you very much to all of you.

**Comment by tristantarrant** [comment, marked as answer]
You need to set it in the `infinispan/server/endpoints/endpoint/rest-connector` configuration. For example, to set it to 20MB:

```xml
<infinispan>
   <server>
      <endpoints>
         <endpoint  socket-binding="default" security-realm="default">
            <hotrod-connector/>
            <rest-connector max-content-length="20000000"/>
         </endpoint>
      </endpoints>
   </server>
</infinispan>
```

**Reply by josemalm32** [reply]
Thank you very much. It worked perfectly. Where in the documentation is that part? I did not find it... 

Cheers and thanks again. 👍

**Reply by tristantarrant** [reply]
It is "documented" in the server XSD, but it should really be in the main docs. I've created https://issues.redhat.com/browse/ISPN-14314 to improve this (and related info).
